### ISSUE TITLE
cncli 6.4.0 crashes: status & sendslots

### ISSUE THREAD

**Issue by rphair** [cardano-community/cncli#65]
`cncli` version `6.4.0` fails for us with an error the program cannot catch.  We assume our DB is OK since `cncli` version `6.3.0` works as expected on the same database.  Running `cncli status` on the newer version:
```
core$ RUST_BACKTRACE=full cncli status --byron-genesis config/byron-genesis.json --shelley-genesis config/shelley-genesis.json --db cncli/cncli.db
thread 'main' panicked at src/nodeclient/leaderlog/mod.rs:762:53:
infallible
stack backtrace:
   0:     0x7f6c613e6535 - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::h1b9dad2a88e955ff
   1:     0x7f6c614f891b - core::fmt::write::h4b5a1270214bc4a7
   2:     0x7f6c613e2f0f - std::io::Write::write_fmt::hd04af345a50c312d
   3:     0x7f6c613e7bb1 - std::panicking::default_hook::{{closure}}::h96ab15e9936be7ed
   4:     0x7f6c613e788c - std::panicking::default_hook::h3cacb9c27561ad33
   5:     0x7f6c60d3aaae - cncli::main::{{closure}}::{{closure}}::h296639538bd52b13
   6:     0x7f6c613e844f - std::panicking::rust_panic_with_hook::hfe205f6954b2c97b
   7:     0x7f6c613e8077 - std::panicking::begin_panic_handler::{{closure}}::h6cb44b3a50f28c44
   8:     0x7f6c613e69f9 - std::sys::backtrace::__rust_end_short_backtrace::hf1c1f2a92799bb0e
   9:     0x7f6c613e7d04 - rust_begin_unwind
  10:     0x7f6c60d1faa3 - core::panicking::panic_fmt::h3d8fc78294164da7
  11:     0x7f6c60d1fa6b - core::option::expect_failed::h3a757a693188cc6e
  12:     0x7f6c60daff6e - cncli::nodeclient::leaderlog::status::h23182a36599ef049
  13:     0x7f6c60d348db - cncli::start::{{closure}}::h5504e8339b303356
  14:     0x7f6c60d3809a - tokio::runtime::park::CachedParkThread::block_on::h62878b35d203bc58
  15:     0x7f6c60d847fc - tokio::runtime::context::runtime::enter_runtime::h4aab6379407dea09
  16:     0x7f6c60d747cb - tokio::runtime::runtime::Runtime::block_on::h4a6215af9f5c4f44
  17:     0x7f6c60d89429 - cncli::main::h1440479ac6ad9c1e
  18:     0x7f6c60d7d5f3 - std::sys::backtrace::__rust_begin_short_backtrace::h8552c05cc54e9d8d
  19:     0x7f6c60d851f9 - std::rt::lang_start::{{closure}}::hc2bf95ae27e769e9
  20:     0x7f6c613dc4b0 - std::rt::lang_start_internal::h5e7c81cecd7f0954
  21:     0x7f6c60d894f5 - main
  22:     0x7f6c61409dba - __libc_start_call_main
  23:     0x7f6c6140b657 - __libc_start_main
  24:     0x7f6c60d20335 - _start
  25:                0x0 - <unknown>
core$
```
Expected: a JSON structure whether the `status` operation was a success or a failure.

OS: Ubuntu 22.04 - `cncli v6.4.0 <d12761b> (x86_64-unknown-linux-gnu)`
 
**Additional context**
`cncli sync` in daemon mode works equally well on versions `6.3.0` and `6.4.0` (though log file format has changed); I haven't tested the other subcommands at this time (generally we verify `"status": "ok"` before running `leaderlog`).

**Comment by cardanoinvest** [comment]
sudo RUST_BACKTRACE=full ./cncli sendslots --byron-genesis /cardano/byron-genesis.json --config config.json --db /mypool/cncli.db --shelley-genesis /cardano/shelley-genesis.json
thread 'main' panicked at src/nodeclient/leaderlog/mod.rs:817:53:
infallible
stack backtrace:
   0:           0x9dc48c - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::h8311dc5f011d295f
   1:           0xa1ede0 - core::fmt::write::hb882977e06b11578
   2:           0x9d8dd0 - std::io::Write::write_fmt::hcf1fde7051011af9
   3:           0x9ddbec - std::panicking::default_hook::{{closure}}::hf76899cf145a898f
   4:           0x9dd7c8 - std::panicking::default_hook::h999a3e46871d878e
   5:           0x42c028 - cncli::main::{{closure}}::{{closure}}::hece6477c66223286
   6:           0x9de3c0 - std::panicking::rust_panic_with_hook::hf138fe657c53ad40
   7:           0x9de020 - std::panicking::begin_panic_handler::{{closure}}::he0a63a4baa3e457d
   8:           0x9dc974 - std::sys::backtrace::__rust_end_short_backtrace::h8cd854c762d85d59
   9:           0x9ddcf0 - rust_begin_unwind
  10:           0x416c9c - core::panicking::panic_fmt::h89cfc268538afcf1
  11:           0x416c6c - core::option::expect_failed::hc3aae2d1e573c88f
  12:           0x490fd4 - cncli::nodeclient::leaderlog::send_slots::h7bba0f8d13135a2e
  13:           0x42775c - cncli::start::{{closure}}::h4245356ae3b7ef8d
  14:           0x42b518 - tokio::runtime::park::CachedParkThread::block_on::hce8c4e1ede21fc21
  15:           0x47775c - tokio::runtime::context::runtime::enter_runtime::h5857b1d49b125d34
  16:           0x45cc88 - tokio::runtime::runtime::Runtime::block_on::h2dcee48dba058cfa
  17:           0x47cec0 - cncli::main::hbf6a2293f2198324
  18:           0x46a968 - std::sys::backtrace::__rust_begin_short_backtrace::h71e528281fe7ccd4
  19:           0x472658 - std::rt::lang_start::{{closure}}::h25d249a41c7d2b7e
  20:           0x9d2230 - std::rt::lang_start_internal::h5e1c352cc23824fa
  21:           0x47cf98 - main

**Comment by AndrewWestberg** [comment]
Closed by: https://github.com/cardano-community/cncli/pull/66

**Comment by rphair** [comment]
Confirmed fixed in [v6.4.1](https://github.com/cardano-community/cncli/releases/tag/v6.4.1) for both `status` and `leaderlog` subcommands.
